include/mbedtls/rsa.h

```diff
@@ -738,39 +775,41 @@ int mbedtls_rsa_rsaes_oaep_decrypt( mbedtls_rsa_context *ctx,
                             size_t output_max_len );
 
 /**
- * \brief          Generic wrapper to perform a PKCS#1 signature using the
- *                 mode from the context. Do a private RSA operation to sign
- *                 a message digest
- *
- * \param ctx      RSA context
- * \param f_rng    RNG function (Needed for PKCS#1 v2.1 encoding and for
- *                               \c MBEDTLS_RSA_PRIVATE)
- * \param p_rng    RNG parameter
- * \param mode     \c MBEDTLS_RSA_PUBLIC or \c MBEDTLS_RSA_PRIVATE
- * \param md_alg   a \c MBEDTLS_MD_XXX (use \c MBEDTLS_MD_NONE for
- *                 signing raw data)
- * \param hashlen  message digest length (for \c MBEDTLS_MD_NONE only)
- * \param hash     buffer holding the message digest
- * \param sig      buffer that will hold the ciphertext
+ * \brief          This function performs a private RSA operation to sign
+ *                 a message digest using PKCS#1. 
+ *
+ *                 It is the generic wrapper for performing a PKCS#1 
+ *                 signature using the \p mode from the context.
+ *
+ * \param ctx      The RSA context.
+ * \param f_rng    The RNG function. Needed for PKCS#1 v2.1 encoding and for
+ *                 #MBEDTLS_RSA_PRIVATE.
+ * \param p_rng    The RNG parameter.
+ * \param mode     #MBEDTLS_RSA_PUBLIC or #MBEDTLS_RSA_PRIVATE.
+ * \param md_alg   An \c MBEDTLS_MD_XXX algorithm. Use #MBEDTLS_MD_NONE for
+ *                 signing raw data.
+ * \param hashlen  The length of the message digest. For #MBEDTLS_MD_NONE only.
```

Comment:
Suggestion: Clarify that `MBEDTLS_MD_NONE` refers to the `md_alg` parameter. Suggestion: `The length of the message digest. Only used if \p md_alg is #MBEDTLS_MD_NONE.`

Reply:
fixed